.jenkins/pipelines/code-coverage.Jenkinsfile

```diff
@@ -5,79 +5,325 @@
+        //     }
+        // }
+        stage('Upload code coverage') {
+            steps {
+                sh """
+                   set -ex
+
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/code-coverage/init-install.sh
+
+                   ${MYST_SCRIPTS}/myst_cc
+                   sed -i 's|SF:${WORKSPACE}|SF:|g' lcov.info
+                   cp lcov.info ${LCOV_FILE}
+
+                   rm -rf ${LCOV_DIR}
+                   mkdir ${LCOV_DIR}
+                   mkdir html_lcov
+
+                   cp -r lcov/* html_lcov/
+                   mv lcov* ${LCOV_DIR}
+                   tar -zcvf ${LCOV_DIR}.tar.gz ${LCOV_DIR}
+                   """
+                
+                publishHTML(target: [
+                    allowMissing: false, 
+                    alwaysLinkToLastBuild: false, 
+                    keepAll: true, 
+                    reportDir: 'html_lcov', 
+                    reportFiles: 'index.html', 
+                    reportName: 'Code Coverage Report', 
+                    reportTitles: ''])
+
+                azureUpload(
+                    containerName: 'mystikos-code-coverage',
+                    storageType: 'container',
+                    uploadZips: true,
+                    filesPath: "${LCOV_DIR}.tar.gz",
+                    storageCredentialId: 'mystikosreleaseblobcontainer'
+                )
+            }
+        }
+        stage('Report code coverage') {
             steps {
-                build job: "Standalone-Pipelines/Measure-Code-Coverage",
+                sh """
+                   lcov --list ${LCOV_FILE} | tee code-coverage-summary
+
+                   tail -n 1 code-coverage-summary | sed 's/|/ /g' | awk '{print "[Line: " \$2 ", Function: " \$4 "] Code Coverage Report"}' > email-subject
+
+                   tail -n 1 code-coverage-summary | sed 's/|/ /g' | awk '{print "Build URL: ${BUILD_INFO}"}' >> email-body
+                   """
+
+                script {
+                    LCOV_REPORT = sh(
+                        returnStdout: true,
+                        script: "cat email-body"
+                    )
+                    EMAIL_SUBJECT = sh(
+                        returnStdout: true,
+                        script: "echo \"\$(cat email-subject) (\$(date '+%Y-%m-%d'))\""
+                    )
+
+                    build job: "/Mystikos/Standalone-Pipelines/Send-Email",
+                    parameters: [
+                        string(name: "REPOSITORY", value: REPOSITORY),
+                        string(name: "BRANCH", value: BRANCH),
+                        text(name: "EMAIL_SUBJECT", value: EMAIL_SUBJECT),
+                        text(name: "EMAIL_BODY", value: LCOV_REPORT)
+                    ]
+                }
+            }
+        }
+        stage('Cleanup') {
+            steps {
+                cleanWs()
+            }
+        }
+    }
+    post {
+        failure {
```

Comment:
Nit: when solutions test failures are caught, this pipeline would send two emails for the same job

Reply:
I removed the report failures step